compiler/src/main/kotlin/io/spine/protodata/CompilerEvents.kt

```diff
@@ -0,0 +1,397 @@
+import com.google.protobuf.Descriptors.FieldDescriptor.Type.ENUM
+import com.google.protobuf.Descriptors.FieldDescriptor.Type.FIXED32
+import com.google.protobuf.Descriptors.FieldDescriptor.Type.FIXED64
+import com.google.protobuf.Descriptors.FieldDescriptor.Type.FLOAT
+import com.google.protobuf.Descriptors.FieldDescriptor.Type.GROUP
+import com.google.protobuf.Descriptors.FieldDescriptor.Type.INT32
+import com.google.protobuf.Descriptors.FieldDescriptor.Type.INT64
+import com.google.protobuf.Descriptors.FieldDescriptor.Type.MESSAGE
+import com.google.protobuf.Descriptors.FieldDescriptor.Type.SFIXED32
+import com.google.protobuf.Descriptors.FieldDescriptor.Type.SFIXED64
+import com.google.protobuf.Descriptors.FieldDescriptor.Type.SINT32
+import com.google.protobuf.Descriptors.FieldDescriptor.Type.SINT64
+import com.google.protobuf.Descriptors.FieldDescriptor.Type.STRING
+import com.google.protobuf.Descriptors.FieldDescriptor.Type.UINT32
+import com.google.protobuf.Descriptors.FieldDescriptor.Type.UINT64
+import com.google.protobuf.Descriptors.FileDescriptor.Syntax
+import com.google.protobuf.Descriptors.OneofDescriptor
+import com.google.protobuf.Empty
+import com.google.protobuf.GeneratedMessageV3.ExtendableMessage
+import io.spine.base.EventMessage
+import io.spine.code.proto.FileSet
+import io.spine.option.OptionsProto
+import io.spine.protobuf.TypeConverter
+import io.spine.protodata.File.SyntaxVersion
+import io.spine.protodata.PrimitiveType.TYPE_BOOL
+import io.spine.protodata.PrimitiveType.TYPE_BYTES
+import io.spine.protodata.PrimitiveType.TYPE_DOUBLE
+import io.spine.protodata.PrimitiveType.TYPE_FIXED32
+import io.spine.protodata.PrimitiveType.TYPE_FIXED64
+import io.spine.protodata.PrimitiveType.TYPE_FLOAT
+import io.spine.protodata.PrimitiveType.TYPE_INT32
+import io.spine.protodata.PrimitiveType.TYPE_INT64
+import io.spine.protodata.PrimitiveType.TYPE_SFIXED32
+import io.spine.protodata.PrimitiveType.TYPE_SFIXED64
+import io.spine.protodata.PrimitiveType.TYPE_SINT32
+import io.spine.protodata.PrimitiveType.TYPE_SINT64
+import io.spine.protodata.PrimitiveType.TYPE_STRING
+import io.spine.protodata.PrimitiveType.TYPE_UINT32
+import io.spine.protodata.PrimitiveType.TYPE_UINT64
+
+/**
+ * A factory for Protobuf compiler events.
+ */
+internal object CompilerEvents {
+
+    /**
+     * Produces a sequence of events based on the given descriptor set.
+     *
+     * The sequence is produced lazily. An element is produced only when polled.
+     *
+     * The resulting sequence is always finite, it's limited by the type set.
+     */
+    fun parse(desc: FileDescriptorSet): Sequence<EventMessage> {
+        val files = FileSet.of(desc.fileList)
+        return sequence {
+            files.files().forEach { produceFileEvents(it) }
+        }
+    }
+
+    /**
+     * Yields compiler events for the given file.
+     *
+     * Opens with an [EnteredFile] event. Then go the events regarding the file metadata. Then go
```

Comment:
I think multiple `Entered...` and `Exited...` events are less convenient in terms of autocompletion and potential keyboard mistype — and less common as to the traditional ES naming — than the approach with the leading subject and trailing participle, such as `FileEntered`.

Please consider switching to a more traditional naming approach.